core/src/main/java/org/apache/calcite/prepare/CalciteCatalogReader.java

```diff
@@ -108,6 +114,23 @@ protected CalciteCatalogReader(CalciteSchema rootSchema,
             : new LinkedHashSet<>(schemaPaths));
     this.typeFactory = typeFactory;
     this.config = config;
+    this.aliasTypeMap = HashBiMap.create();
+    populateAliasTypeMap(this.aliasTypeMap, rootSchema);
+  }
+
+
+  public static void populateAliasTypeMap(BiMap<TypeEntry, String> aliasTypeMap,
```

Comment:
this would be the populating of the type aliases from the schema, it currently doesn't handle non-unique type names yet which needs fixing

Reply:
I'm a bit unsure on whats the types I should be indexing in and out so that we have most versatility when retrieving (I'd imagine doing String <-> String we lose a lot of info)